softdevice_controller/doc/softdevice_controller.rst

```diff
@@ -36,6 +35,54 @@ The following peripherals are owned by the |controller| and must not be accessed
 
 The APIs provided in :file:`sdc_soc.h` and in MPSL provide the application limited access to some of these peripherals.
 
+Message sequence diagrams
+*************************
+
+The message sequence diagram below shows how an example on how to initialize the library.
+Notice that the :ref:`mpsl` must be initialized before the |controller| is initialized.
+The application can choose which features that are enabled and the number of links that are supported.
+
+.. msc::
+    hscale = "1.5";
+    Host,Controller;
+    |||;
+    Host rbox Controller [label = "Library initialization"];
+    Host->Controller      [label="sdc_init()"];
+    Host rbox Controller [label = "Library configuration"];
+    Host->Controller      [label="sdc_cfg_set(SDC_CFG_TYPE_MASTER_COUNT)"];
+    Host->Controller      [label="sdc_cfg_set(SDC_CFG_TYPE_SLAVE_COUNT)"];
+    Host rbox Controller[label = "Feature inclusion"];
+    Host->Controller      [label="sdc_support_adv()"];
+    Host->Controller      [label="sdc_support_slave()"];
+    Host->Controller      [label="sdc_support_2m_phy()"];
+    Host rbox Controller [label = "Enable the SoftDevice Controller"];
+    Host->Controller      [label="sdc_enable()"];
+
+The message sequence diagram below shows a scenario where the application sends a command to the controller.
+Later it retrieves an event and a data packet.
+
+.. msc::
+    hscale = "1.5";
+    Host,Controller;
+    |||;
+    Host rbox Controller [label = "Send a command to the controller"];
+    Host->Controller      [label="sdc_hci_cmd_put()"];
+    Host<-Controller      [label="Host signal is triggered"];
+    Host->Controller      [label="sdc_evt_get()"];
+    Host<<Controller      [label="Command Complete event"];
+    Host rbox Controller [label = "Send data to the controller"];
+    Host->Controller      [label="sdc_hci_data_put()"];
+    Host<-Controller      [label="Host signal is triggered"];
+    Host->Controller      [label="sdc_evt_get()"];
+    Host<<Controller      [label="HCI Number Of Completed packets"];
+    Host rbox Controller [label = "The controller receives some data and raises an event"];
```

Comment:
Should we describe how to accurately use this to prevent event or data stuck in the controller, i.e keep pulling event and data as long as there is at least one available?

Reply:
Sounds reasonable. What about adding: 
_When the host signal is triggered, this indicates that one or more HCI data or event packets are available. Therefore the host should try to pull data and events until it fails._

Reply:
that sounds good.

Reply:
- [ ] Extend docs as suggested.